huaweicloud/services/cbh/data_source_huaweicloud_cbh_flavors.go

```diff
@@ -119,7 +127,18 @@ func datasourceFlavorsRead(_ context.Context, d *schema.ResourceData, meta inter
 	getCBHFlavorsPath := getCBHFlavorsClient.Endpoint + getCBHFlavorsHttpUrl
 	getCBHFlavorsPath = strings.ReplaceAll(getCBHFlavorsPath, "{project_id}",
 		getCBHFlavorsClient.ProjectID)
-	getCBHFlavorsPath += "?action=create"
+
+	// If the `action` is set to **update**, the `spec_code` parameter is required.
+	// If the `action` is not filled in, set it to **create** query.
+	action := d.Get("action")
+	if action != "" {
```

Comment:
```suggestion
	if action, ok := d.GetOk("action"); !ok {
```
the type of function d.Get() result is a interface value, it is not recommand to compare with a string without assert (.string), maybe `d.GetOk()` is better.
The default value `create` in the schema actually is unnecessary